gobblin-utility/src/main/java/gobblin/util/AvroUtils.java

```diff
@@ -417,4 +418,30 @@ public static Schema switchName(Schema schema, String newName) {
     newSchema.setFields(Lists.newArrayList(fieldsNew));
     return newSchema;
   }
+
+  /**
+   * Serialize a generic record as a relative {@link Path}. Useful for converting {@link GenericRecord} type keys
+   * into file system locations. For example {field1=v1, field2=v2} returns field1=v1/field2=v2 if includeFieldNames
+   * is true, or v1/v2 if it is false.
+   * @param record {@link GenericRecord} to serialize.
+   * @param includeFieldNames If true, each token in the path will be of the form key=value, otherwise, only the value
+   *                          will be included.
+   * @return A relative path where each level is a field in the input record.
+   */
+  public static Path serializeAsPath(GenericRecord record, boolean includeFieldNames) {
+    if(record == null) {
+      return new Path("");
+    }
+    List<String> tokens = Lists.newArrayList();
+    for(Schema.Field field : record.getSchema().getFields()) {
+      String sanitizedName = field.name().replaceAll("[\\s/]","_");
```

Comment:
You can define `"[\\s/]"` as a constant. Please also put an empty space before `"_"`.